creator-node/src/middlewares.js

```diff
@@ -144,8 +149,13 @@ async function getOwnEndpoint (req) {
   return spInfo['endpoint']
 }
 
-/** Get all creator node endpoints for user by wallet from discprov. */
-async function getCreatorNodeEndpoints (req, wallet) {
+/**
+ * Get all creator node endpoints for user by wallet from discprov
+ *
+ * If tx blocknumber provided, poll discprov until it has indexed that blocknumber
+ *    Else poll discprov until it returns own endpoint as primary else error
+ */
+async function getCreatorNodeEndpoints (req, wallet, ownEndpoint, ensurePrimary) {
```

Comment:
this function is so hard to read. don't need to rewrite now but it's like so confusing, but it seems correct

Reply:
i actually find this one of the more readable functions but must be because i wrote it lol
anyway, hopefully new comments clarify